message ServerRobot {  
  required fixed32 id = 1;
  optional float velocityx = 2;
  optional float velocityy = 3;
  optional float angle = 4; 
  optional bool hasPuck = 5;
  optional float x = 6;
  optional float y = 7;
  optional bool hasCollided = 8;
  optional fixed32 controllerIp = 9;
  optional fixed32 lastStep = 10;
  optional fixed32 team = 11;
  repeated bool viewLostId = 12; // robot specified by seenById can no 
                                // longer see this robot
  repeated fixed32 seenById = 13; // list of robots now viewing

}
